### PR TITLE
Prevent gamemode pvp.

### PR DESCRIPTION
**Pull request by LegendIsAwesomes**
Title says it all.

https://github.com/TotalFreedom/TotalFreedomMod/issues/287

### CODE REVIEW

**Comment by OxLemonxO** [comment]
I think it should check if the second person is creative or god, then do  nothing if the second person is. I also think that RED would be better then DARK_RED.

**Comment by OxLemonxO** [comment]
Also, this doesn't fully block it. You can still damage the other player with potions.

**Comment by JeromSar** [comment]
@OxLemonxO This PR need not handle that. A separate issue will cover that.

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 1]
```diff
@@ -923,4 +923,32 @@ public void onPlayerLogin(PlayerLoginEvent event)
...
+            }
+            if (event.getDamager() instanceof Arrow)
+            {
+                Arrow arrow = (Arrow) event.getDamager();
+                if (arrow.getShooter() instanceof Player)
+                {
+                    Player player = (Player) arrow.getShooter();
+                    if (player.getGameMode() == GameMode.CREATIVE)
+                    {
+                        TFM_Util.playerMsg(player, "Creative/Godmode pvp is prohibited.", ChatColor.DARK_RED);
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: Inconsistent message.

**Comment by LegendIsAwesomes** [comment]
@JeromSar what do you mean by that, My mind is jumbled up ATM (I'm ill)

**Comment by OxLemonxO** [comment]
@LegendIsAwesomes, the messages are different for the bow hit and the normal hit. Make them the same.

**Comment by LegendIsAwesomes** [comment]
@JeromSar fixed it.

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 2]
```diff
@@ -923,4 +923,32 @@ public void onPlayerLogin(PlayerLoginEvent event)
     {
         TFM_ServerInterface.handlePlayerLogin(event);
     }
+    @EventHandler(priority = EventPriority.HIGHEST)
+    public void onEntityDamageByEntity(EntityDamageByEntityEvent event)
+    {        if (event.getEntity() instanceof Player)
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: Formatting.

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 3]
```diff
@@ -923,4 +923,32 @@ public void onPlayerLogin(PlayerLoginEvent event)
...
+                if (player.getGameMode() == GameMode.CREATIVE)
+                {
+                    TFM_Util.playerMsg(player, "Creative mode pvp is prohibited.", ChatColor.DARK_RED);
+                    event.setCancelled(true);
+                }
+            }
+            if (event.getDamager() instanceof Arrow)
+            {
+                Arrow arrow = (Arrow) event.getDamager();
+                if (arrow.getShooter() instanceof Player)
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: Avoid this nested if-statement here.

**Comment by LegendIsAwesomes** [comment]
@JeromSar what is wrong with nested if-statements.?

**Comment by OxLemonxO** [comment]
Instead of 2 if statements, use &&
@LegendIsAwesomes

**Comment by LegendIsAwesomes** [comment]
@OxLemonxO I don't think you can without nested if-statements being included.

**Comment by JeromSar** [comment]
It's cleaner to avoid nested if-statements, however, upon reconsideration it's not a problem.

**Comment by LegendIsAwesomes** [comment]
@JeromSar it works and has been tested, should be ready to merge.

**Comment by LegendIsAwesomes** [comment]
What's the reason for this to be closed @JeromSar

**Comment by DUFCLiam** [comment]
All PR's where closed see: http://totalfreedom.boards.net/thread/36260/whoops-closed-prs

**Comment by JeromSar** [comment]
Original repository was deleted.

**Comment by LegendIsAwesomes** [comment]
I'll redo this.

**Pull request closed without merging**